bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/BlindHandler.java

```diff
@@ -0,0 +1,184 @@
+
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.library.types.StopMoveType;
+import org.eclipse.smarthome.core.library.types.UpDownType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.lutron.internal.config.BlindConfig;
+import org.openhab.binding.lutron.internal.protocol.LutronCommandType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handler responsible for communicating with Lutron blinds
+ *
+ * @author Bob Adair - Initial contribution based on Alan Tong's DimmerHandler
+ */
+public class BlindHandler extends LutronHandler {
+    private static final Integer ACTION_LIFTLEVEL = 1;
+    private static final Integer ACTION_TILTLEVEL = 9;
+    private static final Integer ACTION_LIFTTILTLEVEL = 10;
+    private static final Integer ACTION_STARTRAISINGTILT = 11;
+    private static final Integer ACTION_STARTLOWERINGTILT = 12;
+    private static final Integer ACTION_STOPTILT = 13;
+    private static final Integer ACTION_STARTRAISINGLIFT = 14;
+    private static final Integer ACTION_STARTLOWERINGLIFT = 15;
+    private static final Integer ACTION_STOPLIFT = 16;
+    private static final Integer ACTION_POSITION_UPDATE = 32; // undocumented in integration protocol guide
+    private static final Integer PARAMETER_POSITION_UPDATE = 2; // undocumented in integration protocol guide
+
+    private int tiltMax = 100; // max 50 for horizontal sheer, 100 for venetian
+
+    private final Logger logger = LoggerFactory.getLogger(BlindHandler.class);
+
+    private BlindConfig config;
+
+    public BlindHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public int getIntegrationId() {
+        if (config == null) {
+            throw new IllegalStateException("handler configuration not initialized");
+        }
+        return config.integrationId;
+    }
+
+    @Override
+    public void initialize() {
+        config = getThing().getConfiguration().as(BlindConfig.class);
+        if (config.integrationId <= 0) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No integrationId configured");
+            return;
+        }
+        if (config.type == null || (!(config.type.equalsIgnoreCase(BLIND_TYPE_SHEER))
+                && !(config.type.equalsIgnoreCase(BLIND_TYPE_VENETIAN)))) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Parameter type not set to valid value");
+            return;
+        }
+        String blindType = config.type;
+        if (blindType.equalsIgnoreCase(BLIND_TYPE_SHEER)) {
```

Comment:
Position literals first in `String` comparisons - that way if the `String` is `null` you won't get a `NullPointerException`, it'll just return `false`. [source: PMD](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#PositionLiteralsFirstInComparisons)